### ISSUE TITLE
IME Plugin Broken

### ISSUE THREAD

**Issue by flippant** [Windower/Issues#820]
IME is broken after the April update. The plugin loads successfully, but the input isn't "queued" for conversion.

**Comment by z16** [comment]
Can you elaborate what you mean, by input not being "queued"? I may be able to check this out later tonight.

**Comment by flippant** [comment]
It was (poor) wording in an effort to describe the sequence of events, in case none of the active developers have used it before.

I've found out that the plugin likely just allows you to use the built-in IME functionality used in the JP client of the game, so my comment is irrelevant anyway!

**Comment by z16** [comment]
So this can be closed?

**Comment by flippant** [comment]
Sorry, no. I meant that the plugin is broken, but I think it isn't actually responsible for the IME interaction in game, it just allows non-JP clients to use the one that's built-in for the JP client. Otherwise, IME support isn't available in the game (Windows IME will lock itself into English).

**Comment by alphaONE2** [comment]
Have you changed your system locale to Japan? FFXI actively switches the IME to romaji input if the system's ANSI codepage is anything other than CP932 (Shift_JIS).

**Comment by flippant** [comment]
Ah, you're right. I can manually switch using the language bar, but the shortcuts don't work.

Either way, it doesn't work correctly. For example, typing "d" in hiragana types "d,,," and the IME immediately changes back to romaji. The text is also never green (or "queued," as I put it).

**Comment by rdbell** [comment]
Does anyone have IME working? Mine is broken too.

**Comment by rdbell** [comment]
It's a year later and I'm still pasting Japanese input into the client because IME is broken. IME works fine in Ashita.

Has anyone looked into fixing this?

**Comment by joshk6656** [comment]
Did you do all this stuff?

http://wiki.windower.net/doku.php?id=plugins:ime

**Comment by rdbell** [comment]
> Did you do all this stuff?
> 
> http://wiki.windower.net/doku.php?id=plugins:ime

Yes and those steps are good for making IME work under Ashita but Windower's IME plugin is still broken.

**Comment by Hoiy** [comment]
I experienced the same issue for over a year as well. I used to be able to input Japanese directly through in-game IME UI with NA client, AppLocale to JP and IME plugin. Then one day this in-game IME no longer shows up. It doesn't work even I switched Windows locale to JP.

Good to know that IME works fine on Ashita, I think this could be fixed here as well(?) I could probably help digging the codebase a bit if anyone could point me to the repository of IME plugin.

**Comment by timsama** [comment]
I went through [these instructions](https://docs.windower.net/plugins/ime2/) and it didn't work. I can get Windows to see that I'm trying to type Japanese into Windower by clicking the language bar, then pressing Ctrl+F2 in game (the Language Bar shows Japanese and Hiragana), but instead of the IME appearing when I try to type "な", [it types "½a" instead](https://i.ibb.co/D7mW4cJ/tried-and-failed.png), and the language bar switches from Hiragana back to Romaji after the first keystroke.

I found that even though my locale was set to Japanese (Japan), my ANSI code page was 65001. This was because I still had [this checkbox](https://i.ibb.co/5BkcF1N/region-settings.png) checked.

Unchecking it and rebooting changed my [ANSI code page to 932](https://i.ibb.co/yRW7MBb/chcp.png), like we want it to be.

But unfortunately, that didn't fix it. It just made it fail [differently](https://i.ibb.co/Dk8SD5j/tried-and-failed2.png) (this was another attempt to type "な", and the immediately-switch-back-to-Romaji issue is still present).

**Comment by sinaraheneba** [comment]
Likely a different issue to what those above are experiencing, but something to be aware of: after happily using IME with no issues over the past year or so, setting up dgvoodoo caused it to start getting stuck inputting alphabet, with input only very briefly restored after restarting Windower entirely. I found the actual cause of the issue was somehow related to Discord's overlay (which dgvoodoo had somehow caused to now function with FFXI), and disabling the overlay in Discord's settings fixed the IME issues for me (and conversely, turning the overlay back on, the issues immediately return).

**Comment by Nifim** [comment]
IME works fine today, It can be hard to setup if you are having issues setting it up come seek assistance in our discord support channel.
